controller/src/main/java/org/jboss/as/controller/operations/global/LocaleResolver.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.jboss.as.controller.operations.global;
+
+import java.util.Locale;
+
+/**
+ * Utility class handling low-level parsing of Locale string tag.
+ *
+ * @author Romain Pelisse - <romain@redhat.com>
+ */
+public final class LocaleResolver {
+
+    private static final String ENGLISH = new Locale("en").getLanguage();
+
+    private LocaleResolver(){};
+
+    static Locale resolveLocale(String unparsed) throws IllegalArgumentException {
+        int len = unparsed.length();
+        if ( len < 1 || len > 7 ) {
```

Comment:
Regarding you previous 3 comments:

1) Yes, a PR for -wildfly-core is the next step. I first fixed the issue in EAP, as there was too much code changes since then to simply apply on Wildfy 9, I opted to have this first reviewed and applied on 8.2.x.

2) Right, would be nice to support , I've file a [WFLY-4239](https://issues.jboss.org/browse/WFLY-4239) about this, and will try to work on it with this fix.

3) Nice sum up, assuming you don't feel disowned of your IP, I'll use part it to the code comment :)